src/toast/_libtoast/ops_pixels_healpix.cpp

```diff
@@ -1,39 +1,62 @@
-// Copyright (c) 2015-2021 by the parties listed in the AUTHORS file.
+// Copyright (c) 2015-2023 by the parties listed in the AUTHORS file.
 // All rights reserved.  Use of this source code is governed by
 // a BSD-style license that can be found in the LICENSE file.
 
-#include <module.hpp>
+// All compiled healpix functionality should be in this file, and it
+// must be a single compilation unit.
 
+#include <module.hpp>
 #include <qarray.hpp>
-
 #include <intervals.hpp>
-
 #include <accelerator.hpp>
 
+// PI
+#define PI 3.14159265358979323846
+
+// 2*PI
+#define TWOPI 6.28318530717958647693
+
 // 2/PI
 #define TWOINVPI 0.63661977236758134308
 
 // 2/3
 #define TWOTHIRDS 0.66666666666666666667
 
-// Helper table initialization
-void hpix_init_utab(uint64_t * utab) {
-    for (uint64_t m = 0; m < 0x100; ++m) {
+const int64_t hpix_jr[] = {2, 2, 2, 2, 3, 3, 3, 3, 4, 4, 4, 4};
+const int64_t hpix_jp[] = {1, 3, 5, 7, 0, 2, 4, 6, 1, 3, 5, 7};
+
+// Helper table initialization, performed on host.
+
+void hpix_init_utab(int64_t * utab) {
+    for (int64_t m = 0; m < 0x100; ++m) {
         utab[m] = (m & 0x1) | ((m & 0x2) << 1) | ((m & 0x4) << 2) |
                   ((m & 0x8) << 3) | ((m & 0x10) << 4) | ((m & 0x20) << 5) |
                   ((m & 0x40) << 6) | ((m & 0x80) << 7);
     }
     return;
 }
 
+void hpix_init_ctab(int64_t * ctab) {
+    for (int64_t m = 0; m < 0x100; ++m) {
+        ctab[m] = (m & 0x1) | ((m & 0x2) << 7) | ((m & 0x4) >> 1) |
```

Comment:
"Hacker's delight" is strong in this one :)

Reply:
That code has been with us since we extended our original healpix implementation from 32bit to 64bit.  No way we should touch it :-)